test/test_bmp.cpp

```diff
@@ -0,0 +1,316 @@
+#include <catch2/catch.hpp>
```

Comment:
I'm not sure what kind of version of Catch2 you are using?  I built one locally from recent `devel` branch of <https://github.com/catchorg/Catch2>, and that doesn't have any `catch2/catch.hpp` include file.  (It has some `catch2/catch_all.hpp`, but when I try with that instead, I get lots of
```
./test_bmp.cpp:49:68: error: ‘Contains’ is not a member of ‘Catch’; did you mean ‘contains’?
   49 |         REQUIRE_THROWS_WITH(not_bmp.read(bmp_path.c_str()), Catch::Contains("Not a BMP") && Catch::Contains("BM"));
      |                                                                    ^~~~~~~~
```
etc. errors.)